doc/userManual.md

```diff
@@ -0,0 +1,183 @@
+# Tcp Server User Manual
+
+## Overview
+
+The `TcpServer` is a module to exchange commands, events, and telemetry with a TCP client, using a TCP/IP connection.
+You can use this module as an interface between a TCP client and a LabVIEW component.
+
+## Design
+
+The `TcpServer` is developed with the Object-Oriented Design (OOD).
+The class diagram is in `doc/uml` directory.
+You can follow [here](../doc/uml/TcpServerClass.uml) for details.
+
+## TcpServer Library
+
+The `TcpServer` module provides a library of VIs for TCP/IP communication.
+It consists of VIs to configure, run and stop the TCP server, and handle the commands, events, and telemetry.
+All of these VIs are public functions of the **TcpServer** class.
+
+The following is a brief description of the main VIs.
+
+## configServer&#46;vi
+
+The configServer VI is used to configure the TCP server, with the port, the number of bytes to read, and other parameters, using the `ServerConfiguration` cluster.
+Clic [here](#server-configuration-cluster) to more details about this cluster.
+
+![configServer](./images/configServer.png)
+
+## runServer&#46;vi
+
+The runServer VI is used to run the TCP server.
+
+![runServer](./images/runServer.png)
+
+## stopServer&#46;vi
+
+The stopServer VI is used to stop all servers.
+
+![stopServer](./images/stopServer.png)
+
+Note:
+
+- Every time you need to stop the TCP server please check if the TCP client is already stopped.
+
+## Using the TcpServer to receive commands
+
+The following is a brief description of the main VIs you have to use to work with **commands** in the TCP server, and an example how to implement the server.
+
+## getInstanceCmd&#46;vi
+
+The getInstanceCmd VI is used to get the `CmdRead` user event instance, to read commands from TCP client.
+
+![getInstanceCmd](./images/getInstanceCmd.png)
+
+## regCmd&#46;vi
+
+The regCmd VI is used to inform the TCP server which commands are registered and allowed to be used by the LabVIEW component.
+Please define all of the registered commands in the `cmdReg` 1-D array of strings.
+
+![regCmd](./images/regCmd.png)
+
+## sendCmdStatus&#46;vi
+
+The sendCmdStatus VI is used to send a `cmdStatus` message to the TCP server to inform the TCP client about the command result.
+You need to add the `sequence_id` of the command received as an input of this function.
+
+![sendCmdStatus](./images/sendCmdStatus.png)
+
+The following image shows an example of a TCP server using the library to receive commands from a TCP client.
+
+Notice that this program configures the TCP server, registers the allowed commands (`move` and `start` in this example), gets the user event reference to receive the commands, runs, and stops the TCP server.
+
+This example registers the `CmdRead` user event and waits for it.
+When the command is received it is showed in the `Command Received` indicator.
+
+![exampleCommand](./images/exampleCommand.png)
```

Comment:
It might be good to mention that the asynchronous call is recommended to run the **runServer.vi** and added the related hyperlink of [Asynchronously Calling VIs](https://zone.ni.com/reference/en-XX/help/371361R-01/lvconcepts/asynchronous_vi_calls/). But the user can also call this vi (**runServer.vi**) directly as a process.

Reply:
Done.